test/dotnet-uninstall.Tests/Shared/Filterers/FiltererTests.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.DotNet.Tools.Uninstall.Shared.BundleInfo;
+using Microsoft.DotNet.Tools.Uninstall.Shared.Configs;
+using Microsoft.DotNet.Tools.Uninstall.Shared.Exceptions;
+using Microsoft.DotNet.Tools.Uninstall.Shared.Filterers;
+using Xunit;
+
+namespace Microsoft.DotNet.Tools.Uninstall.Tests.Shared.Filterers
+{
+    public abstract class FiltererTests
```

Comment:
[Discussed offline; otherwise the filterer names will be ridiculously long, e.g. `AllPreviewsButLatestOptionBundleFilterer`]

Reply:
I think we missed something. I don't think it is "ridiculously long". I think it is a better name since ["honest and complete" is better than "honest"](http://arlobelshee.github.io/Presentations/legacy_code/7_stages_of_naming.png). But I won't block on this issue